src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/BalanceSheetServiceImpl.java

```diff
@@ -12,14 +16,22 @@
     @Autowired
     private BalanceSheetTransformer transformer;
 
+    @Autowired
+    private ApiClientService apiClientService;
+
     @Override
-    public BalanceSheet getBalanceSheet(String transactionId, String companyAccountsId) {
+    public BalanceSheet getBalanceSheet(String transactionId, String companyAccountsId)
+            throws ApiErrorResponseException {
```

Comment:
This is inconsistent with how we are handling errors in the [API services](https://github.com/companieshouse/company-accounts.api.ch.gov.uk/blob/develop/src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImpl.java). We have separated the knowledge bleed between layer so the controller doesn't know the internal behaviour of the service (making an API call).

Reply:
As discussed, we'll address this in a meeting with everyone. I'm of the opinion that we should be following the 'throw early, catch late' principle, but I'm happy to put it to the team